packages/api/src/Module/Module.entity.ts

```diff
@@ -0,0 +1,98 @@
+import { Column, Entity, PrimaryGeneratedColumn, Unique, ManyToOne, JoinColumn } from 'typeorm';
+import { ObjectType, Field, InputType } from '@nestjs/graphql';
+
+import { CMBaseEntity } from '../lib/Base.entity';
+import { Path } from '../Path/Path.entity';
+
+
+export enum ModuleType {
+  assignment = 'assignment',
+  lesson = 'lesson'
+}
+
+@ObjectType()
+export class Module extends CMBaseEntity {
+
+  @Field()
+  id: string;
+
+  @Field()
+  name: string;
+
+  @Field()
+  icon: string;
+
+  @Field()
+  type: ModuleType;
+
+  @Field(() => Module, { nullable: true })
+  previous?: Module;
+
+  @Field(() => Path)
+  path: Path;
+}
+
+@Entity('module')
+@Unique('module_name', ['name'])
+export class EModule extends CMBaseEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column()
+  name: string;
+
+  @Column()
+  icon: string;
+
+  @Column({ type: 'simple-enum', enum: ModuleType })
+  type: ModuleType;
+
+  @ManyToOne(() => EModule, { nullable: true })
+  @JoinColumn({ name: 'previous' })
+  previous: EModule;
+
+  @ManyToOne(() => Path, path => path.module, { nullable: false })
+  path: Path;
+
+}
+
+@InputType()
+export class ModuleInput {
+
+  @Field()
+  name: string;
+
+  @Field()
+  icon: string;
+
+  @Field()
+  type: ModuleType;
+
+  @Field({ nullable: true })
+  previous?: string;
+
+  @Field()
+  path: string;
+}
+
+@InputType()
+export class UpdateModuleInput {
+
+  @Field()
+  id: string;
+
+  @Field({ nullable: true })
+  name?: string;
+
+  @Field({ nullable: true })
+  icon?: string;
+
+  @Field({ nullable: true })
+  type?: ModuleType;
+
+  @Field({ nullable: true })
+  previous?: string;
+
+  @Field({ nullable: true })
+  path?: string;
+}
```

Comment:
You can extend the `ModuleInput` and just add properties that are new

Reply:
The problem here is that in Update the props are optional